app/service/search_index.rb

```diff
@@ -9,35 +9,57 @@ def self.log_elasticsearch_warning(message)
     end
   end
 
-  def initialize(model_name:, id:)
+  def initialize(model_name:, id:, changed: nil)
     @model_name = model_name
     @id = id
+    @changed = changed
   end
 
   def add
-    if search_index_callbacks_enabled?
-      record = model_name.constantize.find(id)
-      record.__elasticsearch__.index_document
-    else
-      log_callback_warning
+    perform { record.__elasticsearch__.index_document }
+  end
+
+  def update
+    return add unless changed.any?
+
+    perform do
+      data = elasticsearch_hash.merge(body: { doc: changes })
+      Elasticsearch::Model.client.update(data)
```

Comment:
Mostly because it wasn't working as expected. Since we're using Sidekiq and we can't pass Ruby objects through it easily (that's why we only pass the `id` and `class`), I wasn't always getting the correct data in ElasticSearch. Going to double-check this and see if I can switch back now that I'm using `after_commit` vs `after_update`.